sdk/core/azure-core/src/http/curl/curl.cpp

```diff
@@ -412,10 +412,18 @@ CURLcode CurlSession::Perform(Context const& context)
     if (hostHeader == headers.end())
     {
       Log::Write(Logger::Level::Verbose, LogMsgPrefix + "No Host in request headers. Adding it");
-      this->m_request.SetHeader("Host", this->m_request.GetUrl().GetHost());
+      std::string headerValue = this->m_request.GetUrl().GetHost();
+      auto port = this->m_request.GetUrl().GetPort();
+      if (port != 0)
+      {
+        headerValue += ":" + std::to_string(port);
+      }
+      this->m_request.SetHeader("Host", headerValue);
     }
-    auto isContentLengthHeaderInRequest = headers.find("content-length");
```

Comment:
I didn't see an explanation of the change to remove the content-length header for HEAD and DELETE.

Why was this change made? Is it a protocol correctness issue? something else?

Reply:
Good catch! I added some explanation in the PR description.